consensus/ibft/signer/helper.go

```diff
@@ -29,7 +29,11 @@ func wrapCommitHash(data []byte) []byte {
 
 // getOrCreateECDSAKey loads ECDSA key or creates a new key
 func getOrCreateECDSAKey(manager secrets.SecretsManager) (*ecdsa.PrivateKey, error) {
-	if !manager.HasSecret(secrets.ValidatorKey) {
+	hasSecret, err := manager.HasSecret(secrets.ValidatorKey)
+	if err != nil {
+		return nil, err
+	}
+	if !hasSecret {
```

Comment:
This might work as expected because `HasSecret` throws error when the key is not saved.